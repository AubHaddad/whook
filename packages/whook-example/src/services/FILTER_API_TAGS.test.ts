import { describe, test, beforeEach, jest, expect } from '@jest/globals';
import initFilterAPITags from './FILTER_API_TAGS.js';
import type { LogService } from 'common-services';

describe('initFilterAPITags', () => {
  describe('should work', () => {
    const log = jest.fn<LogService>();

    beforeEach(() => {
      log.mockClear();
    });

    test('with not ENV', async () => {
      const FILTER_API_TAGS = await initFilterAPITags({
        ENV: {},
        log,
      });

      expect(FILTER_API_TAGS).toMatchInlineSnapshot(`[]`);
      expect({
        logCalls: log.mock.calls,
      }).toMatchSnapshot();
    });

    test('with empty ENV', async () => {
      const FILTER_API_TAGS = await initFilterAPITags({
        ENV: {},
        log,
      });

      expect(FILTER_API_TAGS).toMatchInlineSnapshot(`[]`);
      expect({
        logCalls: log.mock.calls,
      }).toMatchSnapshot();
    });
    test('with some tags in ENV', async () => {
      const FILTER_API_TAGS = await initFilterAPITags({
        ENV: {},
        log,
      });

      expect(FILTER_API_TAGS).toMatchInlineSnapshot(`[]`);
      expect({
        logCalls: log.mock.calls,
      }).toMatchSnapshot();
    });
  });
});
