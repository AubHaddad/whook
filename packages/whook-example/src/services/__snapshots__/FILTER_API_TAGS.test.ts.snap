// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initFilterAPITags should work with empty ENV 2`] = `
{
  "logCalls": [],
}
`;

exports[`initFilterAPITags should work with not ENV 2`] = `
{
  "logCalls": [],
}
`;

exports[`initFilterAPITags should work with some tags in ENV 2`] = `
{
  "logCalls": [],
}
`;
